### ISSUE TITLE
[v1.2.0] client config is missing because soap object is initialized before config

### ISSUE THREAD

**Issue by m-koepke** [savonrb/savon#354]
For some reasons Savon::SOAP::RequestBuilder.soap is accessed before the request_builder.config is set in Savon::Client.
This results in @soap object not having the correct client configuration and thus in the request not having the soap header as it is configured.

Take a look at:
https://github.com/mvladimirovic/savon/commit/e12a7610eecb5edf7b03e61247a94afcb6767f40

I provided a spec that fails without the fix but does not really test the real issue.
Basically it just tests for the symptoms, but I was too lazy to add a real test for the missing soap header.
